libarchive/write.py

```diff
@@ -73,6 +74,39 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry):
```

Comment:
@Changaco changed, I also added `archive_obj_to_write` to ease extending libarchive with custom writers: 9bb50158cc707599f5f974ae4b22c0aa6ce78182